### PR TITLE
Feature/pace l1b converter

### LINKED ISSUE #1061: [Add a converter for PACE radiance data ]
## Description

We add a converter to convert simulated radiance data for PACE mission. This is to convert PACE L1B data. We already implemented a converter for the PACE L2 data

### PR DESCRIPTION
**Pull request by HamidehGMAO**
## Description

We add a converter to convert simulated radiance data for PACE mission. This is to convert PACE L1B data. We already implemented a converter for the PACE L2 data. Data can be obtained from https://oceandata.sci.gsfc.nasa.gov/directdataaccess/Level-1B/PACE_OCI/2019/

### Issue(s) addressed

Link the issues to be closed with this PR

https://github.com/JCSDA-internal/ioda-converters/issues/1061
- fixes #1061

### CODE REVIEW

**Comment by liuxiao37k** [comment]
@HamidehGMAO Thanks for sharing the converter! Looks like `ci` still has some complaints. I'll test later, but today for sure!

**Review by mjhossen — APPROVED** [review]

**Comment by liuxiao37k** [comment]
@HamidehGMAO I tested the batch download shell script with soca-science and it worked perfectly (after replacing USERNAME/PASSWORD part with `--load-cookies ~/.netrc` in the `wget` command). I unfortunately won't be able to test the converter leaving for vacation shortly. Sorry!

A quick question: does the converter loop through all the Level 1b swaths downloaded to the same folder (by soca-science, to `./SCRATCH/2019032112/prep.obs/dwnld/radiance`) and concatenate them to a single converted file like [this Level2 converter](https://github.com/JCSDA-internal/ioda-converters/blob/develop/src/marine/viirs_modis_l2_oc2ioda.py) does? Thank you! I should have looked into the code myself but I am running out of time today...

@mjhossen were you able to test the download script and converter in soca-science by running `prep` step for `EXP_START_DATE=20190321Z12`?

**Comment by HamidehGMAO** [comment]
@liuxiao37k Thanks for looking at the download script. There is a test in this PR ([test/CMakeLists.txt](https://github.com/JCSDA-internal/ioda-converters/pull/1062/files#diff-33394812ba204689144fd2f80832db83853ba1cb32403edb4e15fe4893e675fd)) which test the converter for a sample data "testinput/pace_radiance_L1B.nc" but you can test it for more data.

**Comment by BenjaminRuston** [comment]
@HamidehGMAO are there new data files associated with this PR?

**Comment by HamidehGMAO** [comment]
> @HamidehGMAO are there new data files associated with this PR?

@BenjaminRuston There are just 5 days of simulated data from [21 Mar 2019](https://oceandata.sci.gsfc.nasa.gov/directdataaccess/Level-1B/PACE_OCI/2019/080/) to 25 Mar 2019 available for PACE mission. In this PR we just used a small sample data for 20190321.

File: src/marine/pace_radiance2ioda.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    optional.add_argument(
+        '-t', '--thin',
+        help="percentage of random thinning, from 0.0 to 1.0. Zero indicates"
+             " no thinning is performed. (default: %(default)s)",
+        type=float, default=0.0)
+    optional.add_argument(
+        '--threads',
+        help='multiple threads can be used to load input files in parallel.'
+             ' (default: %(default)s)',
+        type=int, default=1)
```
Comment (travissluka) [inline src/marine/pace_radiance2ioda.py:203]: seems threads aren't used, so remove this

**Review by travissluka — CHANGES_REQUESTED** [review]

File: src/marine/pace_radiance2ioda.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    parser = argparse.ArgumentParser(
+        description=(
+            'Reads the radiance from any PACE L1B Data '
+            ' and converts into IODA formatted output files.'
+            ' Multiple files are concatenated ')
+    )
+    required = parser.add_argument_group(title='required arguments')
+    required.add_argument(
+        '-i', '--input',
+        help="path of patmosx avhrr observation input file(s)",
```
[the remark this answers is not in the document]
Comment (HamidehGMAO) [reply src/marine/pace_radiance2ioda.py]: oops! Just fixed it

File: src/marine/pace_radiance2ioda.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        'nvars': list(chan_number),
+        'ndatetime': list(ndatetime)
+    }
+
+    writer = iconv.IodaWriter(args.output, locationKeyList, DimDict)
+
+    VarAttrs[('radiance', 'ObsValue')]['units'] = 'W m^-2 um^-1 sr^-1'
+    VarAttrs[('radiance', 'ObsError')]['units'] = 'W m^-2 um^-1 sr^-1'
+    VarAttrs[('radiance', 'PreQC')]['units'] = 'unitless'
+    VarAttrs[('radiance', 'ObsValue')]['_FillValue'] = 999
```
[the remark this answers is not in the document]
Comment (HamidehGMAO) [reply src/marine/pace_radiance2ioda.py]: just fixed it. Could you please review it now. Thanks

**Review by kbhargava — APPROVED** [review]

**Review by travissluka — APPROVED** [review]

**Pull request merged**